lua/kulala/parser/headers.lua

```diff
@@ -0,0 +1,85 @@
+--- This class is used to manage the HTTP headers
+--- HTTP header has a case-insensitive name - so class store it in a lower case
+--- and can have a multiple values - which are stored in a table
```

Comment:
The aim is to replace the `kulala/parser/utils.lua` code with this abstraction. I tried to provide a minimal API tailored to kulala's main job converting HTTP specs to curl command line.

At the same the result behaves almost like a normal Lua table, so it's easy to use.